### PR TITLE
Journalist can create article

### PR DESCRIPTION
**Pull request by johanperjulius1**
## This PR contains: 
- Feature tests
- Login functionality
- Displaying logic depending on currentUser role
- 'Create Article' Button and Form

## [PT](https://www.pivotaltracker.com/story/show/175113673)

### CODE REVIEW

File: cypress/integration/journalistCanLoginToCreateArticle.feature.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+      });
+      cy.get('[data-cy="create-article"]').should("not.exist");
+      cy.get('[data-cy="message"]').should(
+        "contain",
+        "Invalid login credentials. Please try again."
+      );
+    });
+  });
+
+  context("login is succesfull but is not a journalist", () => {
```
Comment (oliverochman) [inline cypress/integration/journalistCanLoginToCreateArticle.feature.js:65]: update this test to have the same structure as the other tests

I need to see an assertion for an error message is showed

File: src/components/ArticleForm.jsx [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+  { key: "m", text: "Sports", value: "sports" },
+  { key: "f", text: "Entertainment", value: "entertainment" },
+  { key: "o", text: "Weather", value: "weather" },
+  { key: "o", text: "Business", value: "business" },
+  { key: "o", text: "News", value: "news" },
+];
+
+export default ArticleForm;
+
+// import React, { useState } from "react";
```
Comment (oliverochman) [inline src/components/ArticleForm.jsx]: Clean this file up

File: src/components/Authenticate.jsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+          },
+        });
+
+        history.replace({ pathname: "/" });
+      } else {
+        localStorage.removeItem("J-tockAuth-Storage");
+        setMessage("Sorry, you don't have the necessary permission");
+      }
+    } catch (error) {
+      debugger;
```
Comment (oliverochman) [inline src/components/Authenticate.jsx]: debugger

File: src/components/Authenticate.jsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+          payload: {
+            authenticated: response.success,
+            currentUser: response.data,
+          },
+        });
+
+        history.replace({ pathname: "/" });
+      } else {
+        localStorage.removeItem("J-tockAuth-Storage");
+        setMessage("Sorry, you don't have the necessary permission");
```
Comment (oliverochman) [inline src/components/Authenticate.jsx:32]: this is what I want to see a test for

File: src/modules/articles.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        },
+        {
+          headers: {
+            ...headers,
+            "Content-type": "application/json",
+            Accept: "application/json",
+          },
+        }
+      );
+      debugger;
```
Comment (oliverochman) [inline src/modules/articles.js]: debugger

File: src/modules/articles.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+      document.getElementById("create-article").reset();
```
Comment (oliverochman) [inline src/modules/articles.js:26]: please do this in the component instead

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: src/modules/articles.js (hunk 6 above)
Comment (oliverochman) [reply src/modules/articles.js:26]: ?

**Review by oliverochman — CHANGES_REQUESTED** [review]

**Review by emtalen — APPROVED** [review]

**Pull request merged**